src/fpm_compiler.f90

```diff
@@ -416,12 +416,79 @@ subroutine set_preprocessor_flags (id, flags)
     do i = 1, size(package%preprocess)
         if (package%preprocess(i)%name == "cpp") then
             flags = flag_cpp_preprocessor// flags
+            call get_macros_from_manifest(id, flags, i)
             exit
         end if
     end do
 
 end subroutine set_preprocessor_flags
 
+!> This subroutine will check and get all the macros that are defined in the manifest file
+!> under preprocess section for a specific preprocessor
+subroutine get_macros_from_manifest(id, flags, index_of_preprocessor)
```

Comment:
Is it possible to test this subroutine from fpm's test suite?

Reply:
Added a test for this as well. I had to do a few minor changes in the subroutine to handle the edge cases that were causing the error.